src/dataset.rs

```diff
@@ -935,6 +1013,21 @@ mod tests {
         assert_eq!(ds.raster_count(), 0);
     }
 
+    #[test]
+    fn test_create_layer_options() {
+        use gdal_sys::OGRwkbGeometryType::wkbPoint;
+        let (_temp_path, mut ds) = open_gpkg_for_update(fixture!("poly.gpkg"));
+        let mut options = LayerOptions {
+            name: "new",
+            ty: wkbPoint,
+            ..Default::default()
+        };
+        ds.create_layer(options.clone()).unwrap();
+        assert!(ds.create_layer(options.clone()).is_err());
+        options.options = Some(&["OVERWRITE=YES"]);
+        assert!(ds.create_layer(options.clone()).is_ok());
```

Comment:
Nit: I think you could drop the `clone()` here, just in case `clippy` starts complaining at some point.

Reply:
Good find. Done.